webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/MeshFileService.scala

```diff
@@ -105,43 +170,199 @@ class MeshFileService @Inject()(config: DataStoreConfig)(implicit ec: ExecutionC
         .resolve(meshesDir)
         .resolve(s"${listMeshChunksRequest.meshFile}.$meshFileExtension")
 
-    for {
-      cachedMeshFile <- tryo { meshFileCache.withCache(meshFilePath)(CachedHdf5File.fromPath) } ?~> "mesh.file.open.failed"
-      chunkPositionLiterals <- tryo { _: Throwable =>
-        cachedMeshFile.finishAccess()
-      } {
-        cachedMeshFile.reader
-          .`object`()
-          .getAllGroupMembers(s"/${listMeshChunksRequest.segmentId}/$defaultLevelOfDetail")
-          .asScala
-          .toList
-      }.toFox
-      _ = cachedMeshFile.finishAccess()
-      positions <- Fox.serialCombined(chunkPositionLiterals)(parsePositionLiteral)
-    } yield positions
+    safeExecute(meshFilePath, meshFileCache) { cachedMeshFile =>
+      val chunkPositionLiterals = cachedMeshFile.reader
+        .`object`()
+        .getAllGroupMembers(s"/${listMeshChunksRequest.segmentId}/$defaultLevelOfDetail")
+        .asScala
+        .toList
+      Fox.serialCombined(chunkPositionLiterals)(parsePositionLiteral)
+    }.flatten ?~> "mesh.file.open.failed"
+  }
+
+  def listMeshChunksForSegmentV3(organizationName: String,
+                                 dataSetName: String,
+                                 dataLayerName: String,
+                                 listMeshChunksRequest: ListMeshChunksRequest): Fox[WebknossosSegmentInfo] = {
+    val meshFilePath =
+      dataBaseDir
+        .resolve(organizationName)
+        .resolve(dataSetName)
+        .resolve(dataLayerName)
+        .resolve(meshesDir)
+        .resolve(s"${listMeshChunksRequest.meshFile}.$meshFileExtension")
+
+    safeExecute(meshFilePath, meshFileCache) { cachedMeshFile =>
+      val segmentId = listMeshChunksRequest.segmentId
+      val encoding = cachedMeshFile.reader.string().getAttr("/", "mesh_format")
+      val lodScaleMultiplier = cachedMeshFile.reader.float64().getAttr("/", "lod_scale_multiplier")
+      val transform = cachedMeshFile.reader.float64().getMatrixAttr("/", "transform")
+
+      val (neuroglancerStart, neuroglancerEnd) = getNeuroglancerOffsets(segmentId, cachedMeshFile)
+
+      val manifest = cachedMeshFile.reader
+        .uint8()
+        .readArrayBlockWithOffset("/neuroglancer", (neuroglancerEnd - neuroglancerStart).toInt, neuroglancerStart)
+      val segmentInfo = parseNeuroglancerManifest(manifest)
+      val meshfile = getChunksFromSegmentInfo(segmentInfo, lodScaleMultiplier, neuroglancerStart)
```

Comment:
What exactly is returned here? I feel like meshfile isn’t the perfect name. The type seemst to be MeshSegmentInfo? Also we now have mixed usages of meshfile vs meshFile